### DISCUSSION TITLE
HLK-LD2410B Radar Sensor Support

### DISCUSSION THREAD

**Discussion by Aint-Nobody** [arendst/Tasmota#21727] [Support request]
Hi All,
It would be nice when the onboard light sensitive Sensor would be supported.
So it would be very easy to create a rule on dependency about the light situation.

**Comment by Staars** [comment]
Can you share the technical documentation about the light sensing capabilities of the HLK-LD2410B ?

**Reply by sfromis** [reply]
What I see about light in relation to HLK-LD2410B is not a sensor, but that it is not affected by the surrounding environment, including by light fluctuations. And that they suggest controlling lights as a use case for human presence detection.

**Reply by Aint-Nobody** [reply]
> Can you share the technical documentation about the light sensing capabilities of the HLK-LD2410B ?

It seems that a readout of the value over the serial connection is not supported?
I just wrote a email to the manufacturer to ask how to do read it out over serial.
![IMG-20240704-WA0003](https://github.com/arendst/Tasmota/assets/58624407/676182a9-ce92-410d-a7a2-9cf7b940a9f7)
![IMG-20240704-WA0002](https://github.com/arendst/Tasmota/assets/58624407/588eb07a-16f6-4d7a-9642-1d9948f6cb35)

**Reply by Aint-Nobody** [reply]
> Can you share the technical documentation about the light sensing capabilities of the HLK-LD2410B ?

![IMG_20240704_123018](https://github.com/arendst/Tasmota/assets/58624407/c1a0f8e6-9384-4a50-b77f-b79d227f9dd3)

**Comment by Staars** [comment]
Okay, this thing has some sort of light sensing, but ATM I can not really see the additional benefit to justify the increase of the code size of the driver just for the sake of supporting every imaginable feature.
What is your use case here?

**Reply by Aint-Nobody** [reply]
Well, I have no concrete case, it's only a part of investigation about this sensor.
But let's wait a couple of days if the manufacturer will answer my mail.
If you only need the presence of human function of this sensor, you can use pin 1 of the board so it represents if there is a human detect or not with high or low on this Port. There is no code for any driver needed.
With the ble app you can enable if this Port act on dependency of this light sensor. You can set a threshold and switch if the output works over or under this threshold.

**Comment by joba-1** [comment]
well, I guess on proximity detection switching on light only if it is dark?

**Comment by mr-old-youtube** [comment]
I build a fw witch light sensor of ld2410. You can download at here : http://cloud.mrold.xyz/share/Omy4TcPA
More info can check at my youtube chanel : https://youtu.be/Bmyc6YkT-4Y
